internal/cmd/agent/register.go

```diff
@@ -0,0 +1,87 @@
+package agent
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/sirupsen/logrus"
+	"github.com/spf13/cobra"
+	"k8s.io/client-go/kubernetes"
+	"k8s.io/client-go/rest"
+	ctrl "sigs.k8s.io/controller-runtime"
+	"sigs.k8s.io/controller-runtime/pkg/log"
+	"sigs.k8s.io/controller-runtime/pkg/log/zap"
+
+	command "github.com/rancher/fleet/internal/cmd"
+	"github.com/rancher/fleet/internal/cmd/agent/register"
+	"github.com/rancher/wrangler/v2/pkg/kubeconfig"
+	"github.com/rancher/wrangler/v2/pkg/leader"
+	"github.com/rancher/wrangler/v2/pkg/ratelimit"
+)
+
+func NewRegister() *cobra.Command {
+	cmd := command.Command(&Register{}, cobra.Command{
+		Use:   "register [flags]",
+		Short: "Register agent with an upstream cluster",
+	})
+	return cmd
+}
+
+type Register struct {
+	command.DebugConfig
```

Comment:
This is already in the root command. Since it uses `PersistentPreRun`, it's always executed hence not needed here nor in the other subcommands.